### ISSUE TITLE
[Bug] OOM terminate called after throwing an instance of 'std::bad_alloc' on rtx 5090 cuda 12.8

### ISSUE THREAD

**Issue by pavan098765** [nunchaku-tech/nunchaku#694]
### Checklist

- [x] 1. I have searched for related issues and FAQs (https://nunchaku.tech/docs/nunchaku/faq/faq.html) but was unable to find a solution.
- [x] 2. The issue persists in the latest version.
- [x] 3. Please note that without environment information and a minimal reproducible example, it will be difficult for us to reproduce and address the issue, which may delay our response.
- [x] 4. If your report is a question rather than a bug, please submit it as a discussion at https://github.com/mit-han-lab/nunchaku/discussions/new/choose. Otherwise, this issue will be closed.
- [x] 5. If this is related to ComfyUI, please report it at https://github.com/mit-han-lab/ComfyUI-nunchaku/issues.
- [x] 6. I will do my best to describe the issue in English.

### Describe the Bug

I am on rtx 5090 gpu instance , when I run below code but i get OOM error, i even tried without tea-cache still same error, Any help would be appreciated.

Exact message on terminal:
terminate called after throwing an instance of 'std::bad_alloc'
what():  std::bad_alloc
Aborted

thats it nothing else..

### Environment

Ubuntu 24.04, Python 3,12, Cuda 12.8 Pytorch 2.10

### Reproduction Steps

# flux_kontext_diffusers.py
# Usage: python flux_kontext_diffusers.py
# Dynamic resolution (keeps aspect ratio, multiple-of-64), seed, CFG, steps

import os
import base64
from io import BytesIO
from typing import Dict, Any
from PIL import Image
import torch
torch.cuda.empty_cache()

from diffusers import FluxKontextPipeline
from huggingface_hub import hf_hub_download
from nunchaku import NunchakuFluxTransformer2dModel
from nunchaku.caching.teacache import TeaCache
from nunchaku.utils import get_precision

# -------------------------
# Paths / Model config
# -------------------------
MODEL_DIR = os.path.join("/models_storage", "all_models")
os.makedirs(MODEL_DIR, exist_ok=True)

HF_TOKEN = os.environ.get("HF_TOKEN", "hf_OHFdz")

MODEL_FILES = {
    "DIFFUSION_MODEL": ("nunchaku-tech/nunchaku-flux.1-kontext-dev", f"svdq-{get_precision()}_r32-flux.1-kontext-dev.safetensors"),
}

def download_models():
    local_paths = {}
    for key, (repo_id, filename) in MODEL_FILES.items():
        print(f"[Download] {key} → {repo_id}/{filename}")
        path = hf_hub_download(
            repo_id=repo_id,
            filename=filename,
            token=HF_TOKEN,
            cache_dir=MODEL_DIR,
            resume_download=True
        )
        local_paths[key] = path
    return local_paths

MODEL_PATHS = download_models()
TRANSFORMER_PATH = MODEL_PATHS["DIFFUSION_MODEL"]

# -------------------------
# Initialize Diffusers pipeline
# -------------------------
max_memory = {0:"32GB"}
transformer = NunchakuFluxTransformer2dModel.from_pretrained(TRANSFORMER_PATH, torch_dtype=torch.float16)
pipe = FluxKontextPipeline.from_pretrained(
    "black-forest-labs/FLUX.1-Kontext-dev",
    transformer=transformer,
    torch_dtype=torch.bfloat16,
    device_map="balanced",
    max_memory=max_memory,
)

# Offload + memory optimization
pipe.enable_attention_slicing()
pipe.enable_model_cpu_offload()       # moves unused layers to CPU
pipe.enable_sequential_cpu_offload()  # sequentially offloads layers

# -------------------------
# Helpers
# -------------------------
def _load_image(img_b64: str) -> Image.Image:
    decoded_bytes = base64.b64decode(img_b64)
    return Image.open(BytesIO(decoded_bytes)).convert("RGB")

def _get_dynamic_size(img: Image.Image, max_dim: int = 1152) -> tuple[int, int]:
    w, h = img.size
    aspect = w / h
    if w > h:
        new_w = min(w, max_dim)
        new_h = int(new_w / aspect)
    else:
        new_h = min(h, max_dim)
        new_w = int(new_h * aspect)
    # Multiple-of-64
    new_w = max(64, (new_w // 64) * 64)
    new_h = max(64, (new_h // 64) * 64)
    return new_w, new_h

# -------------------------
# Main inference
# -------------------------
def run_inference_flux_kontext_dev(job_id: str, prompt: str, params: Dict[str, Any]) -> str:
    """
    Run Flux-Kontext inference (image editing) with dynamic resolution using Diffusers.
    Returns the local path to the saved PNG file.
    """
    output_dir = params.get("output_dir", "./outputs")
    os.makedirs(output_dir, exist_ok=True)

    init_images = params.get("images")
    if init_images is None:
        raise ValueError("init_images (base64) is required")
    image = _load_image(init_images)

    # Dynamic resolution
    width, height = _get_dynamic_size(image, max_dim=1152)
    if image.size != (width, height):
        image = image.resize((width, height), resample=Image.LANCZOS)

    # Inference parameters
    steps = params.get("sample_steps", 28)
    cfg = params.get("cfg_scale", 1.5)
    seed = params.get("seed", None)
    if seed is not None and seed < 0:
        seed = None

    generator = torch.Generator(device="cpu").manual_seed(seed) if seed is not None else None

    with TeaCache(model=transformer, num_steps=50, rel_l1_thresh=0.3, enabled=True, model_name="flux-kontext"):
        # Run Diffusers pipeline
        output = pipe(
            image=image,
            prompt=prompt,
            guidance_scale=cfg,
            width=width,
            height=height,
            num_inference_steps=steps,
            generator=generator
        )
    out_image = output.images[0]

    out_path = os.path.join(output_dir, f"flux_kontext_{job_id}.png")
    out_image.save(out_path)
    return out_path

# -------------------------
# Example usage
# -------------------------
if __name__ == "__main__":
    job_id = "tara_dynamic"
    prompt = "Only change the body orientation so that subject is facing the camera directly, with their upper body visible completely from the front. Don't change anything else."
    input_path = "./tara.jpg"

    with open(input_path, "rb") as f:
        img_b64 = base64.b64encode(f.read()).decode("utf-8")

    params = {
        "sample_steps": 30,
        "cfg_scale": 2.5,
        "output_dir": "./outputs",
        "images": img_b64,
        "seed": 42,
    }

    out = run_inference_flux_kontext_dev(job_id, prompt, params)
    print("✅ Output image saved at:", out)

**Comment by lmxyy** [comment]
Is this a jupyter notebook? Could you directly provide it?

**Comment by lmxyy** [comment]
I would suggest not using teacache. Use our FBCache instead.
